File: memory-coder/integrations/bridge_client.js
const BASE_URL = process.env.MEMORY_CODER_BRIDGE_URL || "http://127.0.0.1:3333/v1";

async function post(path, payload) {
  const response = await fetch(`${BASE_URL}${path}`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload)
  });

  const text = await response.text();
  if (!response.ok) {
    throw new Error(text);
  }
  return JSON.parse(text);
}

const command = process.argv[2];
const raw = process.argv[3] || "{}";

if (!command || command === "help" || command === "--help" || command === "-h") {
  console.log(`Memory Coder Bridge Client\n\nUsage:\n  node integrations/bridge_client.js recall '{"query":"cors issue"}'\n  node integrations/bridge_client.js remember '{"content":"...","type":"decision"}'\n`);
  process.exit(0);
}

const routes = {
  "create-project": "/create-project",
  "get-project-context": "/get-project-context",
  "remember": "/remember",
  recall: "/recall",
  "log-bug": "/log-bug"
};

if (!routes[command]) {
  console.error(`Unknown command: ${command}`);
  process.exit(1);
}

post(routes[command], JSON.parse(raw))
  .then((result) => console.log(JSON.stringify(result, null, 2)))
  .catch((error) => {
    console.error(error.message);
    process.exit(1);
  });

File: memory-coder/integrations/bridge_client.py
import json
import sys
import urllib.request

BASE_URL = "http://127.0.0.1:3333/v1"


def post(path, payload):
    data = json.dumps(payload).encode("utf-8")
    request = urllib.request.Request(
        f"{BASE_URL}{path}",
        data=data,
        headers={"Content-Type": "application/json"},
        method="POST",
    )
    with urllib.request.urlopen(request) as response:
        return json.loads(response.read().decode("utf-8"))


def main():
    if len(sys.argv) < 2:
        print("Usage: python bridge_client.py <command> [json]")
        print("Commands: create-project, get-project-context, remember, recall, log-bug")
        sys.exit(1)

    command = sys.argv[1]
    payload = json.loads(sys.argv[2]) if len(sys.argv) > 2 else {}
    routes = {
        "create-project": "/create-project",
        "get-project-context": "/get-project-context",
        "remember": "/remember",
        "recall": "/recall",
        "log-bug": "/log-bug",
    }

    if command not in routes:
        raise SystemExit(f"Unknown command: {command}")

    print(json.dumps(post(routes[command], payload), indent=2, ensure_ascii=False))


if __name__ == "__main__":
    main()
